benchmark/Speed.hs

```diff
@@ -41,7 +39,7 @@ instance Show Model where
   show (HMM xs) = "HMM" ++ show (length xs)
   show (LDA xs) = "LDA" ++ show (length $ head xs)
 
-buildModel :: MonadInfer m => Model -> m String
+buildModel :: MonadInfer n m => Model -> m String
```

Comment:
This extra parameter seems pretty major to me. Can we explain it in the PR itself? I think we should be a bit more disciplined in our PRs going forward (I include myself here and I may be the biggest culprit).